src/js/lib/helpers.js

```diff
@@ -290,4 +290,77 @@
 		}
 	};
 
+	/**
+	 * Finds the root domain
+	 */
+	object.findRootDomain = function () {
+		var cookiePrefix = '_sp_root_domain_test_';
+		var cookieName = cookiePrefix + new Date().getTime();
+
+		var split = window.location.hostname.split('.');
+		var position = split.length - 1;
+		while (position >= 0) {
+			var currentDomain = split.slice(position, split.length).join('.');
+			document.cookie = cookieName + '=;domain=' + currentDomain + ';';
+			if (document.cookie.indexOf(cookieName) > -1) {
```

Comment:
We already have the browser-cookie-lite module as a dependency. You can see it in node_modules/browser-cookie-lite/browser-cookie.js. We might as well use it for setting and getting cookies here.